### DISCUSSION TITLE
The correct way to use permissions to filter data?

### DISCUSSION THREAD

**Discussion by singleseeker** [spatie/laravel-permission#2021] [Q&A] [answered]
For example, 

We have two roles, admin and user, also a model article.

admin -- could get all articles
user -- could get all own articles, all articles whose status is published

What should we do, shell we check the permission in the controller, or add a scope in the model?

**Comment by erikn69** [comment, marked as answer]
Overwrite `checkPermissionTo` adding your custom check
https://github.com/spatie/laravel-permission/blob/6a3ed627cee28a552b5176c172ae0abc5eb30925/src/Traits/HasPermissions.php#L202

**Reply by singleseeker** [reply]
Good job, thank you for ur help.
